### PR TITLE
Set up main branch deployment

### PR DESCRIPTION
**Pull request by t0mmili**
## What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)
Add pipeline for main branch deployment

## What is the new behavior (if this is a feature change)?
See https://github.com/Idea-Pool/js-assertion-cheat-sheet/issues/6

### CODE REVIEW

File: .github/workflows/feature-branch-deployment.yaml [hunk 1]
```diff
@@ -2,9 +2,9 @@ name: Feature Branch Deployment
 
 on:
   workflow_dispatch:
-  push:
-    branches:
-      - '[0-9]+-**'
+#   push:
+#     branches:
+#       - '[0-9]+-**'
```
Comment (kornel-filep) [inline .github/workflows/feature-branch-deployment.yaml:7]: This should be uncommented before merging

File: .github/workflows/main-branch-deployment.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+name: Main Branch Deployment
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+      - '[0-9]+-**'
```
Comment (kornel-filep) [inline .github/workflows/main-branch-deployment.yml:8]: This should be removed from the code before merging. The workflow dispatch as well.

File: .github/workflows/main-branch-deployment.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    paths-ignore:
+      - '**.drawio'
+      - '**.md'
+      - '**.png'
+  pull_request:
```
Comment (kornel-filep) [inline .github/workflows/main-branch-deployment.yml:13]: Is the pull request trigger necessary? I think the on.push.main is enough.

File: .github/workflows/main-branch-deployment.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    branches:
+      - main
+    types: [closed]
+
+env:
+  BRANCH_ID: main
```
Comment (kornel-filep) [inline .github/workflows/main-branch-deployment.yml:19]: I don't see this env variable used anywhere. Is it necessary?

**Review by kornel-filep — CHANGES_REQUESTED** [review]

File: .github/workflows/feature-branch-deployment.yaml (hunk 1 above)
Comment (t0mmili) [reply .github/workflows/feature-branch-deployment.yaml]: It was for testing. Done.

File: .github/workflows/main-branch-deployment.yml (hunk 2 above)
Comment (t0mmili) [reply .github/workflows/main-branch-deployment.yml]: It was for testing. Done.

File: .github/workflows/main-branch-deployment.yml (hunk 3 above)
Comment (t0mmili) [reply .github/workflows/main-branch-deployment.yml]: Looks like it's not needed. Good catch!

File: .github/workflows/main-branch-deployment.yml (hunk 4 above)
Comment (t0mmili) [reply .github/workflows/main-branch-deployment.yml]: It's not relevant anymore.

**Review by kornel-filep — APPROVED** [review]

**Review by szikszail — APPROVED** [review]

**Pull request merged**